Applications/FileIO/GocadIO/GocadAsciiReader.h

```diff
@@ -26,31 +26,44 @@ namespace FileIO
 {
 namespace Gocad
 {
-class GocadTSurfaceReader final
+
+enum class GOCAD_DATA_TYPE
+{
+    UNDEFINED,
+    VSET,
+    PLINE,
+    TSURF,
+    MODEL3D,
+    ALL
+};
+
+class GocadAsciiReader final
 {
 public:
     /**
      * Constructor takes as argument the Gocad .sg text file.
      */
-    explicit GocadTSurfaceReader();
+    explicit GocadAsciiReader();
+
+    explicit GocadAsciiReader(GOCAD_DATA_TYPE const t);
 
-    GocadTSurfaceReader(GocadTSurfaceReader&& src) = delete;
-    GocadTSurfaceReader(GocadTSurfaceReader const& src) = delete;
-    GocadTSurfaceReader& operator=(GocadTSurfaceReader&& rhs) = delete;
-    GocadTSurfaceReader& operator=(GocadTSurfaceReader const& rhs) = delete;
+    GocadAsciiReader(GocadAsciiReader&& src) = delete;
```

Comment:
Why are these `delete`d constructors/assignments needed?

Reply:
c/p from the sgrid reader. I figured it's supposed to be like that since an object copy doesn't really make sense.

Reply:
Just drop them.

Reply:
:heavy_check_mark: